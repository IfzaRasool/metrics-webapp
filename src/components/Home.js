/* eslint-disable jsx-a11y/anchor-is-valid */
// @ts-nocheck
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { RiArrowRightCircleLine } from 'react-icons/ri';
import { displayData } from '../redux/home';
import virusc from './images/virusc.png';

const Home = () => {
  const countryStores = useSelector((state) => state.homeReducer);
  const dispatch = useDispatch();
  return (
    <div>
      <main className="main-container">
        <div className="main">
          <img src={virusc} alt="logo" className="logo" />
          <h2 className="covid-heading">Covid-19 Tracking Cases</h2>
        </div>
        <h2 className="stats-heading">STATS BY COUNTRY</h2>

        <div className="home-container">
          { Object.entries(countryStores).map((e) => (
            <div key={e} id={e[1].id} className="grid-item" onClick={() => dispatch(displayData(e[1].id))}>
              <span className="icon"><RiArrowRightCircleLine /></span>
              <br />
              <Link to="/DetailCountry" className="grid-child-item">{e[0]}</Link>
              <br />
              <span className="cases">
                Total Cases:
                {e[1].today_confirmed}
              </span>
            </div>
          ))}
        </div>
      </main>
    </div>
  );
};

export default Home;
