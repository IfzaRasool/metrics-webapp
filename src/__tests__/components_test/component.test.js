// @ts-nocheck
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import Home from '../../components/Home';

describe('test snapshots for all components', () => {
  const myHome = [];
  it('renders home page correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Home>
            {myHome.forEach(() => renderer.create(<Home />).toJSON())}
          </Home>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
