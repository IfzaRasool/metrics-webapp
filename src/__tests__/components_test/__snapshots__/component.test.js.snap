// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshots for all components renders Detail page correctly 1`] = `
<div>
  <main>
    <div
      className="main-detail-container"
    />
  </main>
</div>
`;

exports[`test snapshots for all components renders home page correctly 1`] = `
<div>
  <main
    className="main-container"
  >
    <div
      className="main"
    >
      <img
        alt="logo"
        className="logo"
        src="virusc.png"
      />
      <h2
        className="covid-heading"
      >
        Covid-19 Tracking Cases
      </h2>
    </div>
    <h2
      className="stats-heading"
    >
      STATS BY COUNTRY
    </h2>
    <div
      className="home-container"
    />
  </main>
</div>
`;

exports[`test snapshots for all components renders home page correctly 2`] = `
<div>
  <main>
    <div
      className="main-detail-container"
    />
  </main>
</div>
`;
